docs/getting-started/creating-plugins.md

```diff
@@ -35,6 +41,53 @@ The `json` file contains the following structure:
 }
 ```
 
+#### Definition structure
+
+The `Definition` property has the following structure:
+
+```json
+{
+  "Id": "Id shall contain a stable identifier for the rule",
+  "Name": "Name may contain an identifier that is understandable to an end user",
+  "HelpUri": "HelUri may contain the absolute URI of the primary documentation for the reporting item",
+  "Message": "Message may contain the default message that will be used",
+  "Kind": "ResultKind of the issue",
+  "Level": "FailureLevel of the issue",
+  "Description": "Description of the rule",
+  "FileNameDenyRegex": "Regular expression that will be denied during analysis",
+  "FileNameAllowRegex": "Regular expression that will be allowed  during analysis"
+}
+```
+
+#### MatchExpression Structure
+
+The `MatchExpression` property has the following structure:
```

Comment:
>MatchExpression [](http://example.com/codeflow?start=5&length=15)

You should describe the inheritance strategy in play  here between definition and match expression